pandas/core/frame.py

```diff
@@ -1476,7 +1476,8 @@ def from_records(cls, data, index=None, exclude=None, columns=None,
 
         return cls(mgr)
 
-    def to_records(self, index=True, convert_datetime64=None):
+    def to_records(self, index=True, convert_datetime64=None,
+                   stringlike_as_fixed_length=False):
```

Comment:
maybe we should instead have a ``dtype=`` parameter? that accepts a dict?

Reply:
In place of the currently added parameter?  That would also work, as then we don’t have to deal with the messy dtype inference ourselves...

I’ll give that implementation a shot.

Reply:
In place of the currently added parameter?  That would also work, as then we don’t have to deal with the messy dtype inference ourselves...

I’ll give that implementation a shot.

Reply:
Because we can specify `dtype` for both columns and indices, added parameters for each.

Also considered just one giant `dtype` parameter to encompass both, but that felt a little cumbersome (e.g. you would always have nested dictionaries), especially people are probably used to generally specifying just column dtypes (e.g. `read_csv`).  Allowing the specification of `index` dtypes is a nice addition (and consistent with the original changes, which affected both indices and columns) nonetheless.